sphinx/ext/todo.py

```diff
@@ -110,9 +110,12 @@ def process_todo_nodes(app, doctree, fromdocname):
 
         for todo_info in env.todo_all_todos:
             para = nodes.paragraph(classes=['todo-source'])
-            description = _('(The <<original entry>> is located in '
-                            ' %s, line %d.)') % \
-                          (todo_info['source'], todo_info['lineno'])
+            if app.config['todo_link_only']:
+                description = _('<<original entry>>')
```

Comment:
Ah, in my case, todo directive is usually used in .rst files. OTOH, your case, todo directive is used in .py files.
I completely agreed your suggestion for `todo directive in .py file`, however it doesn't fit to `todo directive in .rst files`.
It would be better if todolist can generate display relative path and dotted path for .rst and .py.